### PR TITLE
[C-1113] Improve native render performance

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Addresses various areas of the native app that were unnecesarily rerendering. Mainly used `useProxySelector` and `useMemo`

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-1113-improve-app-rerenders

File: packages/common/src/hooks/useSelectTierInfo.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 import { ID } from '../models'
-import { CommonState, makeGetTierAndVerifiedForUser } from '../store'
+import { makeGetTierAndVerifiedForUser } from '../store'
+
+import { useProxySelector } from './useProxySelector'
 
 const getTierAndVerifiedForUser = makeGetTierAndVerifiedForUser()
 
 /**
  * Wraps our reselect tier selector in useMemo and useSelector
```
Comment (sliptype) [inline packages/common/src/hooks/useSelectTierInfo.ts:9]: Our intentions were right haha

File: packages/mobile/src/screens/notifications-screen/Notifications/FavoriteNotification.tsx [hunk 2]
```diff
@@ -29,15 +32,22 @@ type FavoriteNotificationProps = {
 
 export const FavoriteNotification = (props: FavoriteNotificationProps) => {
   const { notification } = props
-  const { userIds, entityType } = notification
-  const users = useSelector((state) =>
-    getNotificationUsers(state, notification, USER_LENGTH_LIMIT)
+  const { id, userIds, entityType } = notification
+
+  if (id === 'f181a99b-b3d8-4607-b874-8a7e95c9284e') {
+    console.log('favorite notif')
```
Comment (sliptype) [inline packages/mobile/src/screens/notifications-screen/Notifications/FavoriteNotification.tsx:38]: Log

File: packages/web/src/common/store/profile/sagas.js [hunk 3]
```diff
@@ -109,18 +109,18 @@ export function* fetchOpenSeaAssets(user) {
   ])
 
   const collectibleList = Object.values(collectiblesMap).flat()
-  if (!collectibleList.length) {
-    console.log('profile has no assets in OpenSea')
```
Comment (sliptype) [inline packages/web/src/common/store/profile/sagas.js:113]: Thanks for getting rid of these!

**Review by sliptype — APPROVED** [review]
Awesome stuff! Some really interesting fixes in here

**Pull request merged**